Sources/Benchmark/BenchmarkReporter.swift

```diff
@@ -12,22 +12,50 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import Dispatch
 import Foundation
 
+#if os(Linux)
+  import Glibc
+#else
+  import Darwin
+#endif
+
 protocol BenchmarkReporter {
-    func report(running name: String, suite: String)
-    func report(results: [BenchmarkResult])
+    mutating func report(running name: String, suite: String)
+    mutating func report(finishedRunning name: String, suite: String)
+    mutating func report(results: [BenchmarkResult])
```

Comment:
I've hit this in my experiments as well. I guess it makes sense to allow for mutating reporters.

Reply:
Ack.